framework/arcane-framework/src/main/scala/services/streaming/IcebergConsumer.scala

```diff
@@ -0,0 +1,101 @@
+package com.sneaksanddata.arcane.framework
+package services.streaming
+
+import models.app.StreamContext
+import models.{ArcaneSchema, DataRow}
+import services.base.SchemaProvider
+import services.lakehouse.{CatalogWriter, SchemaConversions}
+import services.streaming.IcebergConsumer.getTableName
+import services.streaming.base.BatchConsumer
+
+import org.apache.iceberg.rest.RESTCatalog
+import org.apache.iceberg.{Schema, Table}
+import org.slf4j.{Logger, LoggerFactory}
+import zio.stream.{ZPipeline, ZSink}
+import zio.{Chunk, Task, ZIO, ZLayer}
+
+import java.time.format.DateTimeFormatter
+import java.time.{ZoneOffset, ZonedDateTime}
+
+/**
+ * Converts an Arcane schema to an Iceberg schema.
+ */
+given Conversion[ArcaneSchema, Schema] with
```

Comment:
this doesn't belong in the ZIO consumer since its a generic method for translating schemas

Reply:
Discussed by voice: yes, it supposed to be in the `IcebergS3CatalogWriter` file.